python-avd/pyavd/_schema/models/avd_list.py

```diff
@@ -236,3 +236,17 @@ def _cast_as(self, new_type: type[T_AvdList], ignore_extra_keys: bool = False) -
         new_instance._created_from_null = self._created_from_null
 
         return new_instance
+
+    def _compare(self, other: object) -> bool:
+        if not isinstance(other, type(self)):
+            return False
+
+        if len(self) != len(other):
+            return False
+
+        if self._created_from_null != other._created_from_null:
+            return False
```

Comment:
we could revert these two operations as the second one seems less expensive and we could fail faster (same for indexed list)

Reply:
But the likelyhood of _created_from_null is very low.